### PR TITLE
Add missing Linux build dependencies

### PR DESCRIPTION
**Pull request by DocKDE**
I found that builds failed on Arch and OpenSUSE so I added missing dependencies. I also found that OpenSUSE Leap is currently not able to install the required dependencies so I added a check to limit the supported distros to Tumbleweed.

Release Notes:

- N/A

### CODE REVIEW

**Comment by maxdeviant** [comment]
> I also found that OpenSUSE Leap is currently not able to install the required dependencies so I added a check to limit the supported distros to Tumbleweed.

What is the problem you encounter on Leap?

I don’t think we should be adding these sorts of restrictions in the dependency script.

**Comment by DocKDE** [comment]
> What is the problem you encounter on Leap?

I spun up a container from [this](registry.opensuse.org/opensuse/leap:latest) image. On trying to install the dependencies `mold` and `openssl-devel` couldn't be found. I'm not an OpenSUSE guy so I might have missed something here.

> I don’t think we should be adding these sorts of restrictions in the dependency script.

Fair enough, I suppose. I can remove this check again.

File: script/linux [hunk 1]
```diff
@@ -63,11 +63,15 @@ if [[ -n $dnf ]]; then
   exit 0
 fi
 
-# openSUSE
+# openSUSE Tumbleweed
```
Comment (maxdeviant) [inline script/linux:66]:
```suggestion
# openSUSE
```

Comment (DocKDE) [reply script/linux]: Ah, sorry about that.

**Comment by mikayla-maki** [comment]
I believe the CI failure here is spurious: https://github.com/zed-industries/zed/pull/12995

**Pull request merged**